process-dir.js

```diff
@@ -224,6 +240,31 @@ async function process (dir) {
   }))
 }
 
+async function removeEmptyDirectories (dir) {
+  let fileStats = await lstat(dir)
+
+  if (!fileStats.isDirectory()) {
+    return
+  }
+
+  let files = await readdir(dir)
+
+  if (files.length > 0) {
+    let removeEmptyDirectoriesPromises = files.map(
+      file => removeEmptyDirectories(join(dir, file))
+    )
+
+    await Promise.all(removeEmptyDirectoriesPromises)
+
+    // The parent directory may become empty after deleting subdirectories
+    files = await readdir(dir)
```

Comment:
You do not need file system call to detect it, since you remove files in your code. Just change architecture to make it easy to detect what files did you removed